### PR TITLE
easyrsa: use OpenSSL 1.1.x instead of 3.x

### PR DESCRIPTION
**Pull request by whentze**
This version of easyrsa (3.0.8) is not actually compatible with OpenSSL 3.x, support was only added in easyrsa 3.0.9: https://github.com/OpenVPN/easy-rsa/releases/tag/v3.0.9 Using OpenSSL 1.1.x instead fixes several problems with easyrsa, including https://github.com/OpenVPN/easy-rsa/issues/454

I'm proposing to put this right into 22.11 because unstable already has a newer version of easyrsa that's just OpenSSL 3.x compatible.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/easyrsa/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ lib, stdenv, fetchFromGitHub, openssl, runtimeShell }:
+{ lib, stdenv, fetchFromGitHub, openssl_1_1, runtimeShell }:
```
[the remark this answers is not in the document]
Comment (whentze) [reply pkgs/tools/networking/easyrsa/default.nix:1]: Ugh, right. I did it slightly different in my overlay so I wasn't paying attention, sorry.

[the remark this answers is not in the document]
Comment (whentze) [reply pkgs/tools/networking/easyrsa/default.nix:1]: Fixed now :) Thanks

**Comment by ajs124** [comment]
Why can't we just backport the `easyrsa` update?

**Comment by ajs124** [comment]
Ah, sorry, I didn't look closely enough. Since you mentioned 3.0.8 and 3.0.9, I somehow assumed that's what we had on release and master.

Either solution, updating to 3.0.9 or pinning it to `openssl_1_1` on release seems fine to me.

**Comment by whentze** [comment]
I tried upgrading to 3.0.9 naively, but our patches no longer apply to that. Even though it's a patch-level update, the diff is actually huge (and so are the release notes).

**Review by pennae — APPROVED** [review]

**Comment by bobby285271** [comment]
I found this PR accidentally when visiting [actions/workflows/basic-eval](https://github.com/NixOS/nixpkgs/actions/workflows/basic-eval.yml). I haven't really read what is done here, probably you want this to go 22.11 instead of 21.11?

**Comment by whentze** [comment]
New PR at https://github.com/NixOS/nixpkgs/pull/224668

**Pull request merged**